mvpa2/tests/test_giftidataset.py

```diff
@@ -0,0 +1,185 @@
+        nodes = "eJzjYGBgYAdiJiBmBmIAAQAAFQ=="
+
+    nfeatures = 4
+
+    return samples, nodes, nfeatures
+
+
+
+def _get_test_dataset(include_nodes=True):
+    # returns test dataset matching the contents of _get_test_sample_node_data
+    samples, nodes, _ = _get_test_sample_node_data()
+    ds = Dataset(np.asarray(samples))
+
+    if include_nodes:
+        ds.fa['node_indices'] = np.asarray(nodes)
+
+    nsamples = ds.nsamples
+    ds.sa['intents'] = ['NIFTI_INTENT_NONE'] * nsamples
+
+    return ds
+
+
+
+def _build_gifti_string(format_, include_nodes=True):
+    # builds the string contents of a GIFTI file
+    samples, nodes, nfeatures = _get_test_sample_node_data(format_)
+    nsamples = len(samples)
+
+    ndata_arrays = nsamples + (1 if include_nodes else 0)
+
+    header = """<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE GIFTI SYSTEM "http://www.nitrc.org/frs/download.php/115/gifti.dtd">
+<GIFTI Version="1.0"  NumberOfDataArrays="%d">""" % (ndata_arrays)
+
+
+    def _build_data_array(data, intent, encoding=format_, dim=nfeatures):
+        datatype = "INT32" if intent == "NODE_INDEX" else "FLOAT32"
+
+        return """<MetaData/>
+                   <LabelTable/>
+                   <DataArray  ArrayIndexingOrder="RowMajorOrder"
+                               DataType="NIFTI_TYPE_%s"
+                               Dim0="%d"
+                               Dimensionality="1"
+                               Encoding="%s"
+                               Endian="LittleEndian"
+                               ExternalFileName=""
+                               ExternalFileOffset=""
+                               Intent="NIFTI_INTENT_%s">
+                      <MetaData>
+                      </MetaData>
+                      <Data>%s</Data>
+                    </DataArray>
+                """ % (datatype, dim, encoding, intent, data)
+
+
+    body_elements = []
+    if include_nodes:
+        body_elements.append(_build_data_array(nodes, 'NODE_INDEX'))
+
+    body_elements.extend(_build_data_array(sample, 'NONE')
+                         for sample in samples)
+
+    footer = """</GIFTI>"""
+
+    return header + "".join(body_elements) + footer
+
+
+
+@with_tempfile(suffix='.func.gii')
+def test_gifti_dataset(fn):
+    for include_nodes in (False, True):
+        expected_ds = _get_test_dataset(include_nodes)
+
+        expected_ds_sa = expected_ds.copy(deep=True)
+        expected_ds_sa.sa['chunks'] = [4, 3, 2, 1, 3, 2]
+        expected_ds_sa.sa['targets'] = ['t%d' % i for i in xrange(6)]
+
+        for format_ in ("ASCII", "Base64Binary", "GZipBase64Binary"):
+            # build GIFTI file from scratch
+            gifti_string = _build_gifti_string(format_, include_nodes)
+            with open(fn, 'w') as f:
+                f.write(gifti_string)
+
+            # reading GIFTI file
+            ds = gifti_dataset(fn)
+            assert_datasets_almost_equal(ds, expected_ds)
```

Comment:
how do you know that assert_datasets_almost_equal doesn't just not fail always? ;)  i.e. would be nice to have some unittest for it